### PR TITLE
Adding integration test for gardenlet care controller 

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**What this PR does / why we need it**:

Adding integration test for gardenlet care controller 

- Scales down the API Server of the Shoot in the Seed
- Waits for Condition to become unhealthy (Type: APIServerAvailable: false)
- Cleanup to restore state after test

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: test/framework/shootframework.go
Comment (timebertt) [inline test/framework/shootframework.go]: WDYT about adding `interval` and `timeout` as a parameter to `WaitForShootCondition()` to make it more reusable?

File: test/framework/shootframework.go
Comment (timebertt) [inline test/framework/shootframework.go]:
```suggestion
			f.Logger.Infof("Waiting for shoot %s to have expected condition (%s: %s). Currently the condition is not present", f.Shoot.Name, conditionType, conditionStatus)
```

File: test/framework/shootframework.go
Comment (timebertt) [inline test/framework/shootframework.go]:
```suggestion
		f.Logger.Infof("Waiting for shoot %s to have expected condition (%s: %s). Currently: %s: %s", f.Shoot.Name, conditionType, conditionStatus, conditionType, cond.Status)
```

File: test/integration/framework/utils.go
Comment (timebertt) [inline test/integration/framework/utils.go]:
```suggestion
	return ScaleDeployment(setupContextTimeout, client, desiredReplicas, "gardener-scheduler", constants.GardenNamespace)
```

File: test/integration/framework/utils.go
Comment (timebertt) [inline test/integration/framework/utils.go]:
```suggestion
	return ScaleDeployment(setupContextTimeout, client, desiredReplicas, "gardener-controller-manager", constants.GardenNamespace)
```

File: test/integration/shoots/care/shoot_care.go
Comment (timebertt) [inline test/integration/shoots/care/shoot_care.go]:
```suggestion
	timeout = 10 * time.Minute
```

File: test/integration/shoots/care/shoot_care.go
Comment (timebertt) [inline test/integration/shoots/care/shoot_care.go:59]: `*origReplicas` can be `nil` if the test fails on the first assertion.

File: test/integration/shoots/care/shoot_care.go
Comment (timebertt) [inline test/integration/shoots/care/shoot_care.go:74]:
```suggestion
		gomega.Expect(cond).ToNot(BeNil())
		gomega.Expect(cond.Status).To(gomega.Equal(gardencorev1beta1.ConditionTrue))
```

File: test/integration/shoots/care/shoot_care.go
Comment (timebertt) [inline test/integration/shoots/care/shoot_care.go:64]: To me it seems like this does not belong to the test cleanup but rather to the test case itself.
WDYT about moving the deferred code back to the test code and only scale the api server back up in the defer code?
Or is the test expected to leave a healthy Shoot? @dguendisch @schrodit

**Review by timebertt — CHANGES_REQUESTED** [review]

File: test/integration/shoots/care/shoot_care.go
Comment (danielfoehrKn) [reply test/integration/shoots/care/shoot_care.go:59]: Yes, thats why there is a nil check?

Comment (timebertt) [reply test/integration/shoots/care/shoot_care.go:59]: Sure, missed it...

File: test/integration/shoots/care/shoot_care.go
Comment (danielfoehrKn) [reply test/integration/shoots/care/shoot_care.go:64]: As far as I know  to leave a healthy Shoot is the expectation plus I think it makes sense that integration tests should restore the environment.
Otherwise I think the test should be marked as "Destructive"

**Review by timebertt — COMMENTED** [review]
/lgtm now
Waiting for a review by @dguendisch @OlegLoewen @schrodit

**Review by dguendisch — APPROVED** [review]

**Review by rfranzke — APPROVED** [review]

**Pull request merged**
